lib/src/main/java/io/cloudquery/scalar/JSON.java

```diff
@@ -0,0 +1,82 @@
+package io.cloudquery.scalar;
+
+import static io.cloudquery.scalar.ValidationException.NO_CONVERSION_AVAILABLE;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.cloudquery.types.JSONType;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.Arrays;
+import org.apache.arrow.vector.types.pojo.ArrowType;
+
+public class JSON extends Scalar<byte[]> {
+  private static final JSONType dt = new JSONType();
+  private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+  public JSON() {
+    super();
+  }
+
+  public JSON(Object value) throws ValidationException {
+    super(value);
+  }
+
+  @Override
+  protected void setValue(Object value) throws ValidationException {
+    if (value instanceof byte[] bytes) {
+      if (bytes.length == 0) {
+        return;
+      }
+      if (!isValidJSON(bytes)) {
+        throw new ValidationException("invalid json", dt, value);
+      }
+      this.value = bytes;
+    } else if (value instanceof java.lang.String string) {
+      set(string.getBytes());
+    } else {
```

Comment:
We should also allow setting an instance of `JSON` (the scalar `JSON`). We can check if the value is an instance of the class, then set the underlying value directly

Reply:
Actually, I think this scenario is covered in the abstract class [here](https://github.com/cloudquery/plugin-sdk-java/blob/1d31031b08d4784b54dc2a589d9c3b76d3e4f312/lib/src/main/java/io/cloudquery/scalar/Scalar.java#L37-L41)

Reply:
Ah nice, didn't see that. Nice one